src/components/SignUpForm.tsx

```diff
@@ -1,34 +1,50 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { FieldError, useForm } from 'react-hook-form';
 
+import { useAuth } from './hooks/useProvideAuth';
+
 import { Input } from './Input';
 
 export interface ISignUpData {
-  name?: string;
+  name: string;
   email: string;
   password: string;
 }
 
+interface ISignUpForm {
+  onClose: () => void;
+}
+
 export interface IErrorData {
   email?: FieldError | undefined;
   password?: FieldError | undefined;
 }
 
-export const SignUpForm: React.FC = () => {
+export function SignUpForm({ onClose }: ISignUpForm) {
```

Comment:
The name onClose doesn't really reflect what this callback does. I mean here, inside the form, it's called as a result of a successful signup/registration, not as a result of closing something